rustler_mix/lib/rustler/compiler/server.ex

```diff
@@ -0,0 +1,45 @@
+defmodule Rustler.Compiler.Server do
+  use GenServer
```

Comment:
`Rustler.Compiler` is run in parallel by mix, it is required.

Reply:
And actually, we do hold state, the compile artifacts.

Reply:
To elaborate a bit: I ran into the problem of multiple parallel `cargo` invocations on the same thing in the tests, which leads to `cargo` erroring out. Since in the current form in this PR we always run `cargo build` on the whole workspace, by using this server it won't matter and we have all relevant information available to all callers. I tried hard to find a mechanism built into mix to do this but couldn't find anything.

Reply:
:+1: